web/chromecom.js

```diff
@@ -65,7 +64,9 @@ let ChromeCom = {
     file = file.replace(/^drive:/i,
         'filesystem:' + location.origin + '/external/');
 
-    if (/^filesystem:/.test(file) && !PDFJS.disableWorker) {
+    const disableWorker = (window.pdfjsDistBuildPdfWorker &&
```

Comment:
Under which circumstances is `window.pdfjsDistBuildPdfWorker` unset? When the worker script is loaded in the main thread, right? Is it possible for the code to reach this place before the worker script has had a chance to initiate?

In either case, please add a brief comment here to explain what this line is doing..

Reply:
> Under which circumstances is window.pdfjsDistBuildPdfWorker unset? When the worker script is loaded in the main thread, right?

It's unset by default, i.e. when workers *are* being used. Hence it should only ever be defined when the `pdf.worker.js` file is loaded on the main-thread.

> Is it possible for the code to reach this place before the worker script has had a chance to initiate?

That shouldn't happen, since the `resolvePDFFile` function is being called from
https://github.com/mozilla/pdf.js/blob/fe5102a27fac7235e14334664cfc7e6dcaa04a92/web/chromecom.js#L353-L355
which in turn is called from https://github.com/mozilla/pdf.js/blob/fe5102a27fac7235e14334664cfc7e6dcaa04a92/web/app.js#L591-L596
which is called from
https://github.com/mozilla/pdf.js/blob/fe5102a27fac7235e14334664cfc7e6dcaa04a92/web/app.js#L1652-L1654
`webViewerOpenFileViaURL` is then called from
https://github.com/mozilla/pdf.js/blob/fe5102a27fac7235e14334664cfc7e6dcaa04a92/web/app.js#L1536
which is only *after* the viewer initialization code has run, see
https://github.com/mozilla/pdf.js/blob/fe5102a27fac7235e14334664cfc7e6dcaa04a92/web/app.js#L502-L504

Finally, note that we always wait on the hash parameters (among other things) to be initialized in `PDFViewerApplication.initialize`
https://github.com/mozilla/pdf.js/blob/fe5102a27fac7235e14334664cfc7e6dcaa04a92/web/app.js#L166-L172

---

> In either case, please add a brief comment here to explain what this line is doing..

Sure, will do.

*Ninja-edit:* Actually, do we still need this code-path?
This code seem to have been originally added in PR #4598, but the referenced upstream bug http://crbug.com/362061 seem to have been fixed over 3 years ago.
I'd thus assume that the affected versions of Chrome is no longer supported since quite some time, so can we (in a separate PR) remove this code-path and thus not have to worry about it here?

Reply:
I try to not break backcompat unless necessary. I should take a look at my telemetry (https://github.com/Rob--W/pdfjs-telemetry) to see whether it is viable to rip out code that supports old Chrome versions (there is lots of other code besides this line).

Reply:
A comment has now been added here.